### PR TITLE
Select the catalog in the UI

### PR DESCRIPTION
**Pull request by Lenninlasd**
## What does this PR do?

Create two different catalogs with the same backend and reset the whole app search and look at that other catalog.
### Screenshot

image
### Related Issue

[#20](https://github.com/boundlessgeo/exchange-search/issues/20)

### CODE REVIEW

File: src/common/addlayers/ServerService.js [hunk 1]
```diff
@@ -661,8 +666,9 @@ var SERVER_SERVICE_USE_PROXY = true;
     };
 
     this.addSearchResultsForHyper = function(server, filterOptions) {
-      var searchUrl = 'http://geoshape.geointservices.io/search/hypermap/_search?';
+      var searchUrl = service_.catalogList[0].url + '_search?';
       if (filterOptions !== null) {
+        searchUrl = service_.catalogList[filterOptions.catalogKey].url + '_search?';
         searchUrl = service_.applyESFilter(searchUrl, filterOptions);
       }
```
Comment (Lenninlasd) [inline src/common/addlayers/ServerService.js:704]: Hi @milafrerichs, I have some problems with the test, and I don't understand why, can you help me?

File: src/common/addlayers/ServerService.js [hunk 2]
```diff
@@ -668,11 +668,15 @@ var SERVER_SERVICE_USE_PROXY = true;
     this.addSearchResultsForHyper = function(server, filterOptions) {
       var searchUrl = service_.catalogList[0].url + '_search?';
       if (filterOptions !== null) {
-        searchUrl = service_.catalogList[filterOptions.catalogKey].url + '_search?';
+        var catalogKey = filterOptions.catalogKey;
+        if (!isNaN(Number(catalogKey))) {
```
Comment (milafrerichs) [inline src/common/addlayers/ServerService.js]: sorry, this is not the way to fix the failing test.
The failure is the expected outcome. We need to fix the test not check if the catalogKey exist. It does always exist and this needs to blow up if it does not!

**Comment by milafrerichs** [comment]
Have a look at:
https://github.com/terranodo/MapLoom/blob/hypermap/src/common/addlayers/ServerService.spec.js#L213

**Comment by milafrerichs** [comment]
can you explain your changes? what made you change it this particular way?

**Comment by Lenninlasd** [comment]
- I separate `catalogKey` from `filterOptions`, also I validate catalogKey because it could be a wrong value and I test it.
- In the other hand I generalized the pagination result for diferent URL endpoints using `service_.catalogList[catalogKey].url`

**Comment by ingenieroariel** [comment]
Merging for functionality reasons (been open 5 days). Please feel free to continue to the conversation abut the code quality in another issue / pr. Thanks @milafrerichs and @Lenninlasd

**Pull request merged**